Add `role_arn` method to create connection for AWS services (#2672)
astronomer/docs: learn/connections/redshift.md

<TabItem value="iam-creds">

> You can use IAM credentials to connect Airflow to Redshift. This approach lets you use IAM credentials and limits Airflow's permissions. The limitation of this method is that you must mount a credentials file to your Docker containers. 

Reviewer: ```suggestion
You can use IAM credentials to connect Airflow to Redshift. This approach lets you use IAM credentials and limits Airflow's permissions. The limitation of this method is that you must include an AWS credentials file in your Airflow project.
```

> You can use IAM credentials to connect Airflow to Redshift. This approach gives you the option to use IAM credentials and also limit Airflow's permissions. The drawback is the credentials file needs to be mounted to your Docker container. 

Reviewer 2: ```suggestion
You can use IAM credentials to connect Airflow to Redshift. This approach lets you use IAM credentials and limits Airflow's permissions. The limitation of this method is that you must mount a credentials file to your Docker containers. 
```

5. [Generate a new access key ID and secret access key](https://docs.aws.amazon.com/powershell/latest/userguide/pstools-appendix-sign-up.html).

</TabItem>

<TabItem value="role-arn">

> You can use AWS's [Assume Role](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html) method to automatically generate temporary credentials to connect to Redshift. This is useful to grant temporary access to limited privilege AWS IAM users or roles without storing any credentials on disk. Following information is required to create a connection:

Reviewer 2: ```suggestion
You can use AWS's [Assume Role](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html) method to automatically generate temporary credentials to connect to Redshift. This is useful to grant temporary access to limited-privilege IAM users or roles without storing any credentials on disk. The following information is required to create the connection:
```

- Cluster identifier
- Database name
- Port
- Region
- IAM role ARN

Complete the following steps to retrieve these values:

1. In your AWS console, select the region that contains your Redshift cluster, open the Redshift cluster dashboard, then open your cluster. 

2. Open the **General information** tab, then copy the **Cluster identifier** and **Endpoint**.

3. Open the **Properties** tab and copy the **Database name** and **Port**.

> 4. Open your IAM dashboard, and [follow the AWS documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_job-functions_create-policies.html) to create an IAM role and attach an IAM Policy to access the required services, for example AWS Redshift. 

Reviewer 2: ```suggestion
4. Open your IAM dashboard, and follow the [AWS documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_job-functions_create-policies.html) to create an IAM role and attach an IAM policy to access AWS Redshift. 
```
It's a Redshift guide so I assume this is the only resource we need to mention?

> 5. Edit the trust relationship of the role created in step 4 to add a trust policy. For example, the following trust policy allows the IAM role to assume your new role.

Reviewer 2: ```suggestion
5. Edit the trust relationship of the role created in Step 4 to add a trust policy that allows the IAM role to assume your new role.
```
Is it "For example", or do we want users to specifically add a policy for assuming the new role?

Author: We need to add a policy to assume role. I added For example, because the ARN could be different.

Reviewer 2: We would use "for example" if we gave a specific example ARN number. Because this code block uses placeholder values, it's not really an example.

    ![aws-connection-iam-creds](/img/examples/connection-aws-redshift-iam.png)

</TabItem>

<TabItem value="role-arn">

1. Open your Astro project and add the following line to your `requirements.txt` file:

    ```

>     apache-airflow-providers-microsoft-amazon

Reviewer: that's not right :)

    ```

    This will install the Amazon provider package, which makes the Amazon Redshift connection type available in Airflow.

2. Run `astro dev restart` to restart your local Airflow environment and apply your changes in `requirements.txt`.

3. In the Airflow UI for your local Airflow environment, go to **Admin** > **Connections**. Click **+** to add a new connection, then select the connection type as **Amazon Redshift**.

4. Enter a name for the connection in the **Connection Id** field.

> 5. Fill out the following connection fields using the information you retrieved from [Get connection details](#get-connection-details):

Reviewer 2: ```suggestion
5. Complete the following connection fields using the information you retrieved from [Get connection details](#get-connection-details):
```

>     - **Connection Id**: Enter a name for the connection.

Reviewer 2: ```suggestion
```

Author: why did you remove this ? 🤔 I assume by mistake, coz we have this in all the other options

Author: oh! got it. Thanks will remove that step for consistency
